homeassistant/components/kaleidescape/config_flow.py

```diff
@@ -0,0 +1,111 @@
+"""Config flow for Kaleidescape."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, cast
+from urllib.parse import urlparse
+
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.components import ssdp
+from homeassistant.const import CONF_HOST
+
+from . import DeviceInfo, UnsupportedError, validate_host
+from .const import DEFAULT_HOST, DOMAIN, NAME as KALEIDESCAPE_NAME
+
+if TYPE_CHECKING:
+    from homeassistant.data_entry_flow import FlowResult
+
+ERROR_CANNOT_CONNECT = "cannot_connect"
+ERROR_UNKNOWN = "unknown"
+ERROR_UNSUPPORTED = "unsupported"
+
+
+@config_entries.HANDLERS.register(DOMAIN)
+class KaleidescapeConfigFlow(config_entries.ConfigFlow):
+    """Config flow for Kaleidescape integration."""
+
+    VERSION = 1
+
+    discovered_device: DeviceInfo
+
+    async def async_step_user(self, user_input=None) -> FlowResult:
+        """Handle user initiated device additions."""
+        errors = {}
+        host = DEFAULT_HOST
+
+        if user_input is not None:
+            try:
+                info = await validate_host(user_input[CONF_HOST].strip())
+                host = info.host
+
+                if info.server_only:
+                    raise UnsupportedError
+
+                await self.async_set_unique_id(info.serial, raise_on_progress=False)
+                self._abort_if_unique_id_configured(updates={CONF_HOST: host})
+
+                return self.async_create_entry(
+                    title=f"{KALEIDESCAPE_NAME} ({info.name})",
+                    data={CONF_HOST: host},
+                )
```

Comment:
Please move code out of the `try` block that is not expected to raise

Reply:
Fixed.